apps/vaporgui/PMetadataSection.cpp

```diff
@@ -0,0 +1,145 @@
+#include <QTreeWidget>
+#include <QHeaderView>
+#include <sstream>
+
+#include "ErrorReporter.h"
+#include "PMetadataSection.h"
+#include "PSection.h"
+
+#include <vapor/RenderParams.h>
+#include <vapor/DataMgr.h>
+#include <vapor/DC.h>
+
+// clang-format off
+
+namespace {
+    std::vector<std::string> xtypeLookup = { "INVALID", "FLOAT", "DOUBLE", "UINT8", "INT8", "INT32", "INT64", "TEXT" };
+    std::vector<std::string> axisLookup  = { "X", "Y", "Z", "T"};
+}
+
+PMetadataSection::PMetadataSection()
+: PSection("Metadata", { new PMetadataTree() } )
+{
+}
+
+PMetadataTree::PMetadataTree()
+: PWidget("", _tree = new QTreeWidget)
+{
+    _tree->setHeaderHidden(true);
+    _tree->header()->setStretchLastSection(false);
+    _tree->setColumnCount(2);
+    _tree->header()->setSectionResizeMode(QHeaderView::ResizeToContents);
+}
+
+void PMetadataTree::updateGUI() const
+{
+    VAPoR::RenderParams* rp = dynamic_cast<VAPoR::RenderParams*>(getParams());
+    if (rp == nullptr) MSG_ERR("Invalid RenderParams instance for acquiring variable metadata.");
```

Comment:
This throws an error, but then the execution keeps proceeding, doesn't it? I.e. shouldn't you return if on a null pointer?

Reply:
You're right.  Fixed.

Reply:
@sgpearse did you push your fix? I don't see any change here

Reply:
No, not yet.  It's fixed on my local branch and this comment is for my own record keeping.  

There are lot of changes incoming.  There are several caveats I'm running into that I didn't anticipate.  I'll request re-review when ready.